packages/wrangler/src/event-subscriptions/index.ts

```diff
@@ -0,0 +1,376 @@
+				const destination = product
+					? `${s.destination.service}.${s.destination.queue_id}`
+					: JSON.stringify(s.destination);
+				return {
+					id: s.id,
+					name: s.name,
+					source,
+					destination,
+					events: s.events.join(", "),
+				};
+			})
+		);
+	},
+});
+
+export const eventSubscriptionsGetCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "stable",
+		description: "Fetch details about a single Event Subscription",
+	},
+
+	positionalArgs: ["id"],
+	args: {
+		id: {
+			type: "string",
+			description: "ID of the Event Subscription",
+			demandOption: true,
+		},
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+		const { id } = args;
+
+		const subscription = await ctx.fetchResult<EventSubscription>(
+			`/accounts/${accountId}/event_subscriptions/subscriptions/${id}`
+		);
+
+		const destination = `${subscription.destination.service}.${subscription.destination.queue_id}`;
+
+		const product = products(accountId)[subscription.source.service];
+		// If this subscription belongs to a product that is not defined in the
+		// `products()` object, that means we do not definitively know how to
+		// deal with it yet. This will happen when the user is using a Wrangler
+		// version that is older than the version where the product was first
+		// defined. Instead of crashing / ignoring such subscriptions, we'll
+		// try our best to be helpful.
+		const source = product
+			? product.format(subscription.source)
+			: JSON.stringify(subscription.source);
+
+		logger.table([
+			{
+				id: subscription.id,
+				name: subscription.name,
+				source,
+				destination,
+				events: subscription.events.join(", "),
+			},
+		]);
+	},
+});
+
+export const eventSubscriptionsCreateCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "stable",
+		description: "Create a new Event Subscription",
+	},
+
+	positionalArgs: ["name"],
+	args: {
+		name: {
+			type: "string",
+			description: "Name of the subscription",
+			demandOption: true,
+		},
+		source: {
+			type: "string",
+			description:
+				"The service & resource to create the subscription on. Use `event-subscriptions browse` to see available sources and events.",
+			demandOption: true,
+		},
+		destination: {
+			type: "string",
+			description: "ID of the Queue to send events to",
+			demandOption: true,
+		},
+		events: {
+			type: "string",
+			description: "Names of events to subscribe to",
+			demandOption: true,
+		},
+	},
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+
+		const [productId] = args.source.split(".");
+		const product = products(accountId)[productId as EventHubProductId];
+		if (!productId || !product) {
+			throw new CommandLineArgsError(
+				`Unrecognized source service. Must be one of ${Object.keys(products(accountId)).join(", ")}`
```

Comment:
This error might become quite unwieldy once a few more products onboard, should I show the possible options in here?